sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -659,13 +660,19 @@ case class StructsToJson(
         (arr: Any) =>
           gen.write(arr.asInstanceOf[ArrayData])
           getAndReset()
+      case MapType(_: DataType, _: StructType, _: Boolean) =>
+        (map: Any) =>
+          val mapType = child.dataType.asInstanceOf[MapType]
+          gen.write(map.asInstanceOf[MapData], mapType)
+          getAndReset()
     }
   }
 
   override def dataType: DataType = StringType
 
   override def checkInputDataTypes(): TypeCheckResult = child.dataType match {
-    case _: StructType | ArrayType(_: StructType, _) =>
+    case _: StructType | ArrayType(_: StructType, _) |
+         MapType(_: DataType, _: StructType, _: Boolean) =>
```

Comment:
I think this depends how `MapType` is proposed to use here. If we treat a map like a struct similarly, we can support arbitrary map type. Then we need to explicitly clarify it in expression description.